### PR TITLE
feat: add wallet connection button

### PR DESCRIPTION
**Pull request by brolag**

### CODE REVIEW

File: src/hooks/useWallet/index.ts [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+
+      const handleDisconnect = (error: { code: number; message: string }) => {
+        // eslint-disable-next-line no-console
+        console.log("disconnect", error);
+        disconnect();
+      };
+
+      provider.on("accountsChanged", handleAccountsChanged);
+      provider.on("chainChanged", handleChainChanged);
+      provider.on("disconnect", handleDisconnect);
```
Comment (royriver23) [inline src/hooks/useWallet/index.ts:100]: Love these

File: src/lib/chains.ts [hunk 2]
```diff
@@ -0,0 +1,198 @@
...
+
+const supportedChains: IChainData[] = [
+  {
+    name: "Ethereum Mainnet",
+    short_name: "eth",
+    chain: "ETH",
+    network: "mainnet",
+    chain_id: 1,
+    network_id: 1,
+    rpc_url: "https://mainnet.infura.io/v3/%API_KEY%",
```
Comment (royriver23) [inline src/lib/chains.ts:11]: is this supposed to stay like this? `API_KEY`

File: src/lib/utilities.ts [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+  }
+  const chainData = supportedChains.filter(
+    (chain: any) => chain.chain_id === chainId
+  )[0];
+
+  if (!chainData) {
+    throw new Error("ChainId missing or not supported");
+  }
+
+  const API_KEY = "460f40a260564ac4a4f4b3fffb032dad";
```
Comment (royriver23) [inline src/lib/utilities.ts:16]: could this be added to the `constants.ts` too? or is the plan to keep both INFURA_ID AND API_KEY secret later in a .env file?

**Review by royriver23 — APPROVED** [review]
Loved this, added some questions for my own understanding. Thanks, master!

**Review by zeta-o — APPROVED** [review]

**Pull request merged**